### ISSUE TITLE
`Combobox`: `aria-selected` should be set on selected rather than highlighted items

### ISSUE THREAD

**Issue by r52j** [chakra-ui/zag#2947]
### Description

`aria-selected` should be set on the currently selected (i.e. "checked") options within an option list to enable assistive technologies such as screenreaders to grasp the current state of the input (i.e. selection). See for example Ariakit's combobox: https://ariakit.org/examples/combobox-multiple. When using a screenreader, for example, options with `aria-selected` set will be announced as "selected" (tested in Chrome with VoiceOver on macOS).

As opposed to that, Ark sets `aria-selected` on the currently highlighted (i.e. "focused") option, which is meaningless as it means that when navigating through the option list using the arrow keys, *all* items will be announced as selected and thus, the users has no way of grasping the actual selection.

### Link to Reproduction (or Detailed Explanation)

https://ark-ui.com/docs/components/combobox#examples

### Steps to Reproduce

1. Open the dropdown of the combobox
2. Open your browser's inspector and navigate to markup of the options within the popover
3. Highlight different options using the mouse or error keys and observe how the attachment of `aria-selected` changes

### Ark UI Version

5.30.0

### Framework

- [x] React
- [ ] Solid
- [ ] Vue
- [ ] Svelte

**Comment by segunadebayo** [comment]
I just pushed a fix for this. We'll release an update shortly.

If the issue persists after upgrading, I'll re-open it.
